### PR TITLE
Update monitoring

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

File: resourcemonitor/resourcemonitor.go [hunk 1]
```diff
@@ -335,177 +319,281 @@ func (monitor *ResourceMonitor) StopInstanceMonitor(instanceID string) error {
 	return nil
 }
 
+func (monitor *ResourceMonitor) GetAverageMonitoring() (cloudprotocol.NodeMonitoringData, error) {
```
Comment (mlohvynenko) [inline resourcemonitor/resourcemonitor.go:322]: Add documentation comment

File: resourcemonitor/alertprocessor.go [hunk 2]
```diff
@@ -49,34 +64,77 @@ func createAlertProcessor(name string, source *uint64,
 func (alert *alertProcessor) checkAlertDetection(currentTime time.Time) {
 	value := *alert.source
 
-	if value >= alert.rule.High && alert.thresholdTime.IsZero() {
-		log.WithFields(log.Fields{
-			"name": alert.name, "maxThreshold": alert.rule.High, "value": value,
-		}).Debugf("Max threshold crossed")
+	if !alert.alertCondition {
```
Comment (mlohvynenko) [inline resourcemonitor/alertprocessor.go:67]: consider splitting this logic into 2 private methods

**Review by mlohvynenko — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

File: resourcemonitor/resourcemonitor.go (hunk 1 above)
Comment (al1img) [reply resourcemonitor/resourcemonitor.go:323]: done

File: resourcemonitor/alertprocessor.go (hunk 2 above)
Comment (al1img) [reply resourcemonitor/alertprocessor.go:67]: done

**Pull request merged**